cms/djangoapps/contentstore/api/views.py

```diff
@@ -0,0 +1,206 @@
+            course_dir = path(settings.GITHUB_REPO_ROOT) / base64.urlsafe_b64encode(repr(courselike_key))
+            temp_filepath = course_dir / filename
+            if not course_dir.isdir():  # pylint: disable=no-value-for-parameter
+                os.mkdir(course_dir)
+
+            log.debug('importing course to {0}'.format(temp_filepath))
+            with open(temp_filepath, "wb+") as temp_file:
+                for chunk in request.FILES['course_data'].chunks():
+                    temp_file.write(chunk)
+
+            log.info("Course import %s: Upload complete", courselike_key)
+            with open(temp_filepath, 'rb') as local_file:
+                django_file = File(local_file)
+                storage_path = course_import_export_storage.save(u'olx_import/' + filename, django_file)
+
+            async_result = import_olx.delay(
+                request.user.id, text_type(courselike_key), storage_path, filename, request.LANGUAGE_CODE)
+            return Response({
+                'task_id': async_result.task_id
+            })
+        except Exception as e:
+            return self.make_error_response(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                developer_message=str(e),
+                error_code='internal_error'
+            )
+
+    """
+    **Use Case**
+
+        * Get a status on an asynchronous task import
+
+    **Example Request**
+
+        GET /api/courses/v0/import/course-v1:edX+DemoX+Demo_Course/
+
+    **GET Parameters**
+
+        A GET request must include the following parameters.
+
+        * task_id: (required) The UUID of the task to check
+
+    **GET Response Values**
+
+        If the import task is found successfully by the UUID provided, an HTTP
+        200 "OK" response is returned.
+
+        The HTTP 200 response has the following values.
+
+        * state: String description of the state of the task
+
+
+    **Example GET Response**
+
+        {
+            "state": "Succeeded"
+        }
+
+    """
+    def get(self, request, course_id):
+        """
+        Check the status of the specified task
+
+        Args:
+            request (Request): Django request object.
+            course_id (string): URI element specifying the course location.
+            task_id (string): URI element specifying the course location.
+
+        Return:
+            state (string): State of the task, one of the UserTaskState statuses
+        """
+
+        courselike_key = CourseKey.from_string(course_id)
+        if not has_course_author_access(request.user, courselike_key):
+            return self.make_error_response(
+                status_code=status.HTTP_403_FORBIDDEN,
+                developer_message='The user requested does not have the required permissions.',
+                error_code='user_mismatch'
+            )
+        try:
+            task_id = request.GET['task_id']
+            task_status = UserTaskStatus.objects.get(task_id=task_id)
```

Comment:
There's no check that the task_id corresponds to the course_id. Is that OK? I could sneak around authentication by naming a course I have access to, and a task_id from a different course (though I don't know how I would get the task_id, or what sensitive info is in the status of the task).